### PR TITLE
User's own XYZ support + smaller enhancements

### PR DESCRIPTION
**Pull request by blokhin**

### CODE REVIEW

File: src/components/include/DefaultImportModalDialog.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+            </ModalHeader>
+        )
+    }
+
+    renderBody() {
+        return (
+            <ModalBody className="bgm-dark">
+                <BasisText
+                    ref={(el) => {this.BasisTextComponent = el}}
+                    className="combinatorial-basis"
```
Comment (timurbazhirov) [inline src/components/include/DefaultImportModalDialog.js:149]: "default-import-modal-basis-text" perhaps?

File: src/components/include/DefaultImportModalDialog.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                    onChange={this.handleChange}
+                />
+                <input type="file"
+                    id="fileapi"
+                    onChange={(event) => this.handleFileChange(event.target.files)}
+                />
+
+                <div className="row m-t-10">
+                    <div className="col-md-12">
+                        <button id="generate-combinatorial" className="btn btn-custom btn-block"
```
Comment (timurbazhirov) [inline src/components/include/DefaultImportModalDialog.js:160]: "default-import-modal-submit"?

File: src/components/include/DefaultImportModalDialog.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    handleFileChange(files) {
+        if (!files[0] || !files[0].size) return alert("Error: file cannot be read (unaccessible?)");
+        this.reader.currentFilename = files[0].name;
+        this.reader.readAsText(files[0]);
+    }
+
+    renderHeader() {
+        return (
+            <ModalHeader className="bgm-dark" closeButton={true}>
+                <h4 className="modal-title">{this.props.title}
```
Comment (timurbazhirov) [inline src/components/include/DefaultImportModalDialog.js:131]: We should have a proper title and remove the link on the lines below - the combinatorial basis concept needs more explanation, here we don't need it

File: src/components/include/DefaultImportModalDialog.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        basisConfig.coordinates.forEach(function(item){
+            x.push(item.value[0]);
+            y.push(item.value[1]);
+            z.push(item.value[2]);
+        });
+
+        const diffX = _.max(x) - _.min(x),
+            diffY = _.max(y) - _.min(y),
+            diffZ = _.max(z) - _.min(z);
+        const mockLatticeConstant = _.max([diffX, diffY, diffZ]) * mockLatticeConstantMultiplicator;
```
Comment (timurbazhirov) [inline src/components/include/DefaultImportModalDialog.js:88]: Lines 75-88 should go elsewhere. We should probably move this to Made, maybe to `utils` or `Basis` there.

File: src/components/include/DefaultImportModalDialog.js [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+import {Made} from "@exabyte-io/made.js";
+import {ModalHeader, ModalBody} from 'react-bootstrap';
+
+
+import {Material} from "../../material";
+import {ModalDialog} from './ModalDialog';
+import BasisText from "../source_editor/BasisText";
+
+
+export const defaultXYZMaterialConfig = {
```
Comment (timurbazhirov) [inline src/components/include/DefaultImportModalDialog.js:12]: We should move this to Made.js - somewhere close to where Xyz parsers are, perhaps.

**Comment by timurbazhirov** [comment]
A couple more things in addition to the inline comments above:

(1) the pull request should be into "dev" branch, not master directly
(2) we should add tests for the new functionality and assert them passing locally

@blokhin

**Comment by blokhin** [comment]
Alright. I'll return to that as I have a little bit of free time.

**Pull request closed without merging**